### PR TITLE
docs: refactor sidebar label translations examples to use tags with regional subtags

### PR DESCRIPTION
**Pull request by HiDeoo**
#### What kind of changes does this PR include?

- Changes or translations of Starlight docs site content

#### Description

This pull request is an attempt to improve the documentation regarding the translation of sidebar labels following [this issue](https://discord.com/channels/830184174198718474/1070481941863878697/1160916855318401075) on Discord. The scenario is the following:

- A user wants to translate some sidebar labels to some language, in this case, Brazilian Portuguese.
- After checking the [sidebar guide](https://starlight.astro.build/guides/sidebar/#internationalization) and the [Configuration Reference](https://starlight.astro.build/reference/configuration/#translating-labels), they use the following configuration:

```js
{
  label: 'Getting Started',
  translations: { 'pt-br': 'Introdução' },
  link: '/getting-started',
}
```

Altho, this configuration does not translate the sidebar link label due to the fact that the key of the `translations` object should use a BCP-47 tag as key, which would be `pt-BR` in this case while the user thought it should be `pt-br` and match their file system structure.

To improve this, this PR proposes the following changes:

- Sidebar guide
  - Explicitely mention that a BCP-47 language tag should be used as key with some examples
  - Update the example to translate to Brazilian Portuguese instead of Spanish
- Configuration Reference
  - Update the example to translate to Brazilian Portuguese instead of French

### CODE REVIEW

**Review by delucis — APPROVED** [review]
This is great — thank you @HiDeoo!

Let’s hold off before merging this — I’d love to get a few outstanding translation PRs in before.

File: docs/src/content/docs/guides/sidebar.mdx
Comment (HiDeoo) [reply docs/src/content/docs/guides/sidebar.mdx:363]: It feels like that if we link to something, it should be a complete list and this one is only a small subset of commonly used primary language subtags (they even say less than 2 percent of primary language subtags). 

Considering the [official registry](https://www.iana.org/assignments/language-subtag-registry/language-subtag-registry) is not the "most user-friendly", maybe we could link to https://r12a.github.io/app-subtags/ ?

Any thoughts? Or maybe the wikipedia one is enough?

Comment (delucis) [reply docs/src/content/docs/guides/sidebar.mdx:363]: That’s the app I use regularly, but has the drawback of not providing much context. https://www.w3.org/International/questions/qa-choosing-language-tags might be more handy if you’re looking to learn what BCP-47 means?

Comment (kevinzunigacuellar) [reply docs/src/content/docs/guides/sidebar.mdx:363]: Great point! As a user following this guide, I am thinking about learning more about what BPC-47 means. However, the following line explains it very well with the example. 

So it would make sense to share a website where you can enter a language and you can find out the BPC-47 for that language.

I think the github page link would be best.

Comment (HiDeoo) [reply docs/src/content/docs/guides/sidebar.mdx:363]: > [w3.org/International/questions/qa-choosing-language-tags](https://www.w3.org/International/questions/qa-choosing-language-tags) might be more handy if you’re looking to learn what BCP-47 means?

Great page, didn't know about this one. I also just noticed that https://r12a.github.io/app-subtags/ is actually linked in it as the "Language Subtag Lookup tool" so considering this, I think I may slightly prefer the W3C one for a guide too. 🤷

Comment (delucis) [reply docs/src/content/docs/guides/sidebar.mdx:363]: Feel free to choose what you think is best @HiDeoo!

Comment (HiDeoo) [reply docs/src/content/docs/guides/sidebar.mdx]: Went with the W3C one as it also links to the other one.

Thanks Kevin for the suggestion of adding a link and Chris for that link 🙌 If I got that right, everyone should get their co-author credits.

**Review by delucis — APPROVED** [review]
Lovely! Merge whenever you like @HiDeoo 🌟

**Pull request merged**
